mergify_cli/ci_issues/utils.py

```diff
@@ -0,0 +1,47 @@
+import pathlib
+
+import httpx
+
+from mergify_cli import console
+from mergify_cli import utils
+
+
+async def raise_for_status(response: httpx.Response) -> None:
+    if response.is_error:
+        await response.aread()
+        details = response.text or "<empty_response>"
+        console.log(f"[red]Error details: {details}[/]")
+
+    response.raise_for_status()
+
+
+def get_ci_issues_client(
+    api_url: str,
+    token: str,
+) -> httpx.AsyncClient:
+    return utils.get_http_client(
+        api_url,
+        headers={
+            "Authorization": f"Bearer {token}",
+        },
+        event_hooks={
+            "request": [],
+            "response": [raise_for_status],
+        },
+    )
+
+
+def get_files_to_upload(
```

Comment:
this is only used once in upload so put it there?
especially since this just uploads XML file blindly so it really is not "utils" at all.